### DISCUSSION TITLE
Undeclared VM object with Lesson 7 deployment

### DISCUSSION THREAD

**Discussion by RichardPalladino** [Cyfrin/foundry-full-course-cu#172] [Q&A] [answered]
I tried to resolve this with Phind and other resources but nothing has worked so far.

I am running Foundry/Forge in a WSL environment.  Everything else has worked so far without issue (at least I haven't received any Foundry/Forge specific errors).

When I write the code to deploy FundeMe.sol for lesson 7, the `vm` object is not recognized.  I tried running `foundryup` and `forge init solidity scripting` , without success.  I also tried some solutions the AI's gave me, like importing the Vm object directly from 'forge-std/Vm.sol'

Do I just need to start with a new workspace?

![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/120539891/2543eaac-3750-41ee-b97d-ffb6bb78e71c)

**Comment by pyromaus** [comment, marked as answer]
try adding `contract DeployFundMe is Script` to line 12

**Comment by RichardPalladino** [comment]
I'm so glad I asked here.  That solved the issue.

Fail early and often, I guess, to learn it well.

After reading your answer, I thought, "Well that's obvious."  But this just means I won't forget it in the future. Thank you.

**Comment by pyromaus** [comment]
> Fail early and often, I guess, to learn it well.

@RichardPalladino lmao yeah bro 100%. It took me several fails of forgetting the `is` before I learned to habitually make sure it's there if needed. 
Like half the time you have an error, is little stuff like this man lol. Never hesitate to ask on here - someone was probably in your shoes!! Cheers
